### ISSUE TITLE
tf.data.Dataset 训练数据中断

### ISSUE THREAD

**Issue by ChenShisen** [Sanster/tf_crnn#10]
print('Begin training...')
        for epoch in range(self.epoch_start_index, self.cfg.epochs):  
            self.sess.run(self.tr_ds.init_op)  #问题

当训练图片文件夹里图片少时 可以跑的通，当生产百万的图片，程序就在#问题 处停了，没有报错
这种读取方式是把图片全部读到内存里吗 还是按批次在硬盘读数据，请教一下，谢谢

**Comment by Sanster** [comment]
是把图片的绝对路径读到内存里面，初始化的时候 tensorflow 的 dataset 会对所有路径 shuffle 一次，是会比较慢。

**Comment by ChenShisen** [comment]
我的直接结束了训练进程，也没有报错，就直接停了，换一个训练集较小的文件夹，又可以正常训练，找不到原因，也不抛出异常，请问您知道是什么原因吗？

**Comment by ChenShisen** [comment]
谢谢作者 找到原因了  内存不够 关闭一些程序就跑通了 谢谢

**Comment by rechardchen123** [comment]
> 是把图片的绝对路径读到内存里面，初始化的时候 tensorflow 的 dataset 会对所有路径 shuffle 一次，是会比较慢。

作者你好，我最近在跑的时候也遇到了这个问题，看你的回答是一次shuffle完全部数据集，每次读入的现存的数据量太大，导致显存空间不足。修改shuffle的参数可以解决这个问题吗？我报的错误如下：
ResourceExhaustedError (see above for traceback): OOM when allocating tensor with shape[2512,1200001] and type float on /job:localhost/replica:0/task:0/device:GPU:0 by allocator GPU_0_bfc
         [[node bilstm_2/fully_connected/MatMul (defined at /home/ucesxc0/anaconda3/envs/text_renderer/lib/python3.6/site-packages/tensorflow/contrib/layers/python/layers/layers.py:1854)  = MatMul[T=DT_FLOAT, transpose_a=false, transpose_b=false, _device="/job:localhost/replica:0/task:0/device:GPU:0"](bilstm_2/Reshape, bilstm_2/fully_connected/weights/read)]]
Hint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info.

         [[{{node Adadelta/update/_156}} = _Recv[client_terminated=false, recv_device="/job:localhost/replica:0/task:0/device:CPU:0", send_device="/job:localhost/replica:0/task:0/device:GPU:0", send_device_incarnation=1, tensor_name="edge_3356_Adadelta/update", tensor_type=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:CPU:0"]()]]
Hint: If you want to see a list of allocated tensors when OOM happens, add report_tensor_allocations_upon_oom to RunOptions for current allocation info.
